### PR TITLE
Add namespacetypes endpoint

### PR DESCRIPTION
**Pull request by vjuux**
## Description

Add `/namespacetypes` endpoint which returns the type of the current namespace (passed in the X-Namespace header).

### CODE REVIEW

File: .spelling [hunk 1]
```diff
@@ -113,6 +113,7 @@ namespace
 NamespaceMappingPolicy
 NamespaceRenderer
 NamespaceUnique
+NamespaceType
```
Comment (ericrpowers) [inline .spelling:116]: Should be above `NamespaceUnique`

File: specs/namespacetype.spec [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Model
+model:
+  rest_name: namespacetype
+  resource_name: namespacetypes
+  entity_name: NamespaceType
+  package: squall
+  group: core/namespace
+  description: |-
+    Returns the type of the current namespace (passed in the X-Namespace header).
```
Comment (ericrpowers) [inline specs/namespacetype.spec:9]:
```
Returns the type of the specified namespace.
```
You can drop the stuff in the parentheses as that is the only way to pass in the namespace. Also, for formatter does not like the use of `|-` in the description for me.

File: specs/namespacetype.spec [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+# Attributes
+attributes:
+  v1:
+  - name: type
+    description: Type of namespace.
+    type: string
```
Comment (ericrpowers) [inline specs/namespacetype.spec:16]: You may want to make this an `enum` like namespace has for its type and use the same `allowed_choices` (I noticed that your branch is a bit behind so merge in master for the latest types).

File: specs/root.spec [hunk 4]
```diff
@@ -636,6 +636,10 @@ relations:
   create:
     description: Renders the namespace where an object should reside.
 
+- rest_name: namespacetype
+  get:
+    description: Returns the type of the current namespace (passed in the X-Namespace header).
```
Comment (ericrpowers) [inline specs/root.spec:641]:
```
Returns the type of the specified namespace.
```

**Review by ericrpowers — COMMENTED** [review]
I think you missed committing `namespacetype.go`?

File: specs/namespacetype.spec (hunk 2 above)
Comment (vjuux) [reply specs/namespacetype.spec]: I've seen this `|-` in multiple places, wondering why would it trip over this

**Comment by vjuux** [comment]
@ericrpowers This should be better now :p

File: specs/namespacetype.spec [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    type: enum
+    exposed: true
+    read_only: true
+    autogenerated: true
+    transient: true
+    allowed_choices:
+    - Default
+    - Tenant
+    - CloudAccount
+    - Group
```
Comment (ericrpowers) [inline specs/namespacetype.spec:31]: This is an old namespace type. You need to pull in latest master for the revised types (and updated description).

File: specs/namespacetype.spec (hunk 2 above)
Comment (t00f) [reply specs/namespacetype.spec]: `|-` is used for multiline text

File: specs/namespacetype.spec [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+  resource_name: namespacetypes
+  entity_name: NamespaceType
+  package: squall
+  group: core/namespace
+  description: Returns the type of the specified namespace.
+
+# Attributes
+attributes:
+  v1:
+  - name: type
```
Comment (t00f) [inline specs/namespacetype.spec:13]: `type` is a keyword in golang, let's name it `namespaceType`.

You don't need an enum here, just put a string that is readonly.
* No default value, no transient.
* readonly, autogenerated

**Review by t00f — CHANGES_REQUESTED** [review]

File: specs/namespacetype.spec (hunk 6 above)
Comment (vjuux) [reply specs/namespacetype.spec:13]: It seems that I'm missing something here, don't understand why we're using type `string` over `enum` with default value :/

**Review by t00f — APPROVED** [review]
LGTM, thanks @vjuux !

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "namespacetype",
    "component": "gaia",
    "pr-id": "566",
    "commit-sha": "b6990769d4459fa1584d379c7e39f4758f296fe1",
    "pipeline": "master"
  },
  {
    "project": "namespacetype",
    "component": "backend",
    "pr-id": "903",
    "commit-sha": "dcec287c8320167e3069da11c20ab449397bcf11",
    "pipeline": "master"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "namespacetype",
    "component": "gaia",
    "pr-id": "566",
    "commit-sha": "22ef074416cc7f3ab7869060950ac02eb66765f6",
    "pipeline": "master"
  },
  {
    "project": "namespacetype",
    "component": "backend",
    "pr-id": "903",
    "commit-sha": "dcec287c8320167e3069da11c20ab449397bcf11",
    "pipeline": "master"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "namespacetype",
    "component": "gaia",
    "pr-id": "566",
    "commit-sha": "39dc7d9384413502ab63ba19ee4e94b427360fd9",
    "pipeline": "master"
  },
  {
    "project": "namespacetype",
    "component": "backend",
    "pr-id": "903",
    "commit-sha": "dcec287c8320167e3069da11c20ab449397bcf11",
    "pipeline": "master"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "namespacetype",
    "component": "backend",
    "pr-id": "903",
    "commit-sha": "1c76f7da1f6d98e5fb3ca295c6ac6669244c66d5",
    "pipeline": "master"
  },
  {
    "project": "namespacetype",
    "component": "gaia",
    "pr-id": "566",
    "commit-sha": "39dc7d9384413502ab63ba19ee4e94b427360fd9",
    "pipeline": "master"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "namespacetype",
    "component": "gaia",
    "pr-id": "566",
    "commit-sha": "39dc7d9384413502ab63ba19ee4e94b427360fd9",
    "pipeline": "master"
  },
  {
    "project": "namespacetype",
    "component": "backend",
    "pr-id": "903",
    "commit-sha": "db74446bf134275143849b4fb0f1c4bb032a6d07",
    "pipeline": "master"
  }
]
```

**Review by t00f — APPROVED** [review]
LGTM

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "namespacetype",
    "component": "gaia",
    "pr-id": "566",
    "commit-sha": "39dc7d9384413502ab63ba19ee4e94b427360fd9",
    "pipeline": "master"
  },
  {
    "project": "namespacetype",
    "component": "backend",
    "pr-id": "903",
    "commit-sha": "db74446bf134275143849b4fb0f1c4bb032a6d07",
    "pipeline": "master"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "namespacetype",
    "component": "backend",
    "pr-id": "903",
    "commit-sha": "7d41faf69bb26128d649f8fcb73a5d2f9b2364f9",
    "pipeline": "master"
  },
  {
    "project": "namespacetype",
    "component": "gaia",
    "pr-id": "566",
    "commit-sha": "39dc7d9384413502ab63ba19ee4e94b427360fd9",
    "pipeline": "master"
  }
]
```

**Comment by t00f** [comment]
/build - automatically fired by gogo with following PRs and commit SHAs v1.0.0
```
[
  {
    "project": "namespacetype",
    "component": "gaia",
    "pr-id": "566",
    "commit-sha": "39dc7d9384413502ab63ba19ee4e94b427360fd9",
    "pipeline": "master"
  },
  {
    "project": "namespacetype",
    "component": "backend",
    "pr-id": "903",
    "commit-sha": "c544f642a6c1a573e6e650d6fd6d6f6c8264e857",
    "pipeline": "master"
  }
]
```

**Pull request merged**